core/src/main/java/com/linecorp/armeria/client/retry/RetryingRpcClient.java

```diff
@@ -124,9 +172,16 @@ private void doExecute0(ClientRequestContext ctx, RpcRequest req,
         final RpcResponse res = executeWithFallback(unwrap(), derivedCtx,
                                                     (context, cause) -> RpcResponse.ofFailure(cause));
 
+        final RetryConfig<RpcResponse> retryConfig = mapping().get(ctx, req);
+        if (!retryConfig.needsContentInRule()) {
+            logger.warn("RetryingRpcClient is being used with RetryRule (without content).");
+        }
+        final RetryRuleWithContent<RpcResponse> retryRule =
+                retryConfig.needsContentInRule() ?
+                retryConfig.retryRuleWithContent() : RetryRuleUtil.fromRetryRule(retryConfig.retryRule());
```

Comment:
In my opinion
* It would be noisy to leave a warn message every time.
* It is not efficient to convert `RetryRule` to `RetryRuleWithContent` everytime.

How about adding a package-private method to `RetryConfig` such as 
```java
class RetryConfig {
	RetryRuleWithContent<T> fromRetryRule() {
	    if (fromRetryRule == null) {
	      logger.warn("...")
	      fromRetryRule = RetryRuleUtil.fromRetryRule(retryConfig.retryRule());
	    }
	    return fromRetryRule;
	}
}
```

Reply:
Good call! Done.